extensions/scarb-doc/tests/test.rs

```diff
@@ -1,117 +1,44 @@
 
-        /// Main function that calculates the 16th Fibonacci number
-        fn main() -> u32 {
-            fib(16)
-        }
-
-        /// use into_trait
-        use core::traits::Into as into_trait;
-        use core::traits::TryInto;
-
-        /// FOO constant with value 42
-        const FOO: u32 = 42;
-
-        /// Calculate the nth Fibonacci number
-        ///
-        /// # Arguments
-        /// * `n` - The index of the Fibonacci number to calculate
-        ///
-        fn fib(mut n: u32) -> u32 {
-            let mut a: u32 = 0;
-            let mut b: u32 = 1;
-            while n != 0 {
-                n = n - 1;
-                let temp = b;
-                b = a + b;
-                a = temp;
-            };
-            a
-        }
-
-        /// Pair type alias for a tuple of two u32 values
-        type Pair = (u32, u32);
-
-        /// Color enum with Red, Green, and Blue variants
-        enum Color {
-            /// Red color
-            Red: (),
-            /// Green color
-            Green: (),
-            /// Blue color
-            Blue: (),
-        }
-
-        /// Shape trait for objects that have an area
-        trait Shape<T> {
-            /// Constant for the shape type
-            const SHAPE_CONST: felt252;
+const ROOT_PACKAGE_NAME: &str = "hello_world";
+const SUB_PACKAGE_NAME: &str = "hello_world_sub_package";
 
-            /// Type alias for a pair of shapes
-            type ShapePair;
-
-            /// Calculate the area of the shape
-            fn area(self: T) -> u32;
-        }
+const FEATURE_NAME: &str = "test_feature";
 
-        /// Circle struct with radius field
-        #[derive(Drop, Serde, PartialEq)]
-        struct Circle {
-            /// Radius of the circle
-            radius: u32,
-        }
-
-        /// Implementation of the Shape trait for Circle
-        impl CircleShape of Shape<Circle> {
-            /// Type alias for a pair of circles
-            type ShapePair = (Circle, Circle);
-
-            /// Shape constant
-            const SHAPE_CONST: felt252 = 'xyz';
-
-            /// Implementation of the area method for Circle
-            fn area(self: Circle) -> u32 {
-                3 * self.radius * self.radius
-            }
-        }
-
-        /// Tests module
-        mod tests {
-            /// Imported fib function from the parent module
-            use super::fib as fib_function;
-
-            /// Really
-            #[test]
-            /// works.
-            fn it_works() {
-                assert(fib_function(16) == 987, 'it works!');
-            }
-        }
-    "#
-};
+const FIBONACCI_CODE_WITHOUT_FEATURE: &str = include_str!("code/code_1.cairo");
+const FIBONACCI_CODE_WITH_FEATURE: &str = include_str!("code/code_4.cairo");
+const COMMON_CODE_WITHOUT_FEATURE: &str = include_str!("code/code_2.cairo");
+const COMMON_CODE_WITH_FEATURE: &str = include_str!("code/code_3.cairo");
```

Comment:
Personally, I am not a big fan of test fixtures like this, as I am not sure we will reuse it across multiple test cases and the names do not help either. If I was doing it, I would probably just `indoc` them here directly, maybe splitting features tests cases into a separate file (and thus also the code - hiding it there from my eyes). 
However, it's only me, so I am open to leave it if you have a contrary opinion :D

Reply:
Imo ur right about hiding it under a certain variable, but this way we are separating them, because they are very large. Just wanted to make a test.rs file just about testing logic, not about certain "snapshots". We could also separate different test regarding features in a multiple different files, but I think they are so much connected to each other. Putting them in separate files would make it just unclear. Imho I would leave it like that, because there is so much code in those snapshots, that scrolling up to see what the code actually do and finding the example would take more time, than peaking inside the snapshot code file